libraries/standard/mqtt/src/mqtt_state.c

```diff
@@ -323,23 +544,246 @@ static MQTTStatus_t addRecord( MQTTPubAckInfo_t * records,
     return status;
 }
 
+/*-----------------------------------------------------------*/
+
 static void updateRecord( MQTTPubAckInfo_t * records,
                           size_t recordIndex,
                           MQTTPublishState_t newState,
                           bool shouldDelete )
 {
+    uint16_t packetId;
+
+    assert( records != NULL );
+
     if( shouldDelete == true )
     {
+        /* Mark the record as invalid. */
         records[ recordIndex ].packetId = MQTT_PACKET_ID_INVALID;
-        records[ recordIndex ].qos = MQTTQoS0;
-        records[ recordIndex ].publishState = MQTTStateNull;
```

Comment:
What's wrong with keeping this? I think it makes the records cleaner

Reply:
We don't need that. We are just depending on packet ID for determining if a record is present or not. When a new record is added, we are anyway updating all the fields. Hence I don't think this clean up is required. It came up in a discussion with Dan.